File: 268 Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int res = nums.length;					// a XOR a = 0, 0 XOR a = a, 且XOR具交換律，故將全部都XOR後留下的即為落單的數。
        for(int i = 0; i < nums.length; i++)
            res ^= i ^ nums[i];
        return res;
    }
}

/* Another solution:
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int total = n * (n + 1) / 2;
        int sum = 0;
        for(int num : nums) {
            sum += num;
        }
        return total - sum;
    }
}
*/

File: 268 Missing Number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        res = 0
        n = len(nums)
        for i in range(n):
            res ^= i ^ nums[i]
        res ^= n
        return res
''' Faster solution using the sum difference
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        return n*(1+n)//2 - sum(nums)
'''